lib/stages/RfiFrameDrop.cpp

```diff
@@ -284,21 +316,22 @@ bool RfiFrameDrop::rest_thresholds_callback(nlohmann::json& update) {
             return false;
         }
 
-        thresholds_new->thresholds.push_back(
-            {threshold, (size_t)(fraction * sk_samples_per_frame), fraction});
+        thresholds_new.push_back({threshold, fraction});
+        num_sk.push_back((size_t)(fraction * sk_samples_per_frame));
     }
 
+
     // Resize sk_exceeds to length of thresholds
-    thresholds_new->sk_exceeds.resize(thresholds_new->thresholds.size(), 0);
+    sk_exceeds.resize(thresholds_new.size(), 0);
 
     INFO("Setting new RFI excision cuts:");
-    for (auto& [threshold, t, fraction] : thresholds_new->thresholds) {
-        (void)t;
-        INFO("  added cut with threshold={}, fraction={}", threshold, fraction);
-    }
+    for (size_t i = 0; i < thresholds_new.size(); i++)
+        INFO("  added cut with threshold={}, fraction={}", thresholds_new[i].first,
+             thresholds_new[i].second);
```

Comment:
Why don't you use the structured binding range based for that was previously used?
```suggestion
for (auto& [threshold, fraction] : thresholds_new->thresholds)
    INFO("  added cut with threshold={}, fraction={}", threshold, fraction);
```

Reply:
Also they could be `const auto&`.